cluster-autoscaler/cloudprovider/gce/mig_info_provider.go

```diff
@@ -151,6 +153,23 @@ func (c *cachingMigInfoProvider) getCachedMigForInstance(instanceRef GceRef) (Mi
 func (c *cachingMigInfoProvider) RegenerateMigInstancesCache() error {
 	c.cache.InvalidateAllMigInstances()
 	c.cache.InvalidateAllInstancesToMig()
+	c.cache.InvalidateMigInstancesState()
+
+	if c.bulkGceMigInstancesListingEnabled {
+		err := c.fillMigInfoCache()
+		if err != nil {
+			return err
+		}
+		instances, listErr := c.listInstancesInAllZonesWithMigs()
+		migToInstances := groupInstancesToMigs(instances)
+		updateErr := c.updateMigInstancesCache(migToInstances)
+
+		if listErr != nil {
+			return listErr
+		}
+		return updateErr
+	}
+
 	migs := c.migLister.GetMigs()
 	errors := make([]error, len(migs))
 	workqueue.ParallelizeUntil(context.Background(), c.concurrentGceRefreshes, len(migs), func(piece int) {
```

Comment:
Should this be skipped if bulkGceMigInstancesListingEnabled=true?

Reply:
Yes. We are doing the same thing in different ways.